src/backend/distributed/master/master_repair_shards.c

```diff
@@ -350,11 +449,12 @@ SearchShardPlacementInList(List *shardPlacementList, char *nodeName, uint32 node
 
 /*
  * CopyShardCommandList generates command list to copy the given shard placement
- * from the source node to the target node.
+ * from the source node to the target node. Caller could optionally skip copying
+ * the data by the flag includeDataCopy.
  */
 List *
-CopyShardCommandList(ShardInterval *shardInterval,
-					 char *sourceNodeName, int32 sourceNodePort)
+CopyShardCommandList(ShardInterval *shardInterval, char *sourceNodeName,
+					 int32 sourceNodePort, bool includeDataCopy)
```

Comment:
`copyData` would look better (to me at least)

Reply:
This is directly coming from enterprise repo. So, I didn't want to diverge. (Btw, this will be merge conflict anyway)